File: DotNET/Endpoint Examples/JSON Payload/signed-pdf-non-pfx.cs

using Newtonsoft.Json;
using Newtonsoft.Json.Linq;
using System.Text;

namespace Samples.EndpointExamples.JsonPayload
{
    public static class SignedPdfNonPfx
    {
        public static async Task Execute(string[] args)
        {
            if (args == null || args.Length < 3)
            {
                Console.Error.WriteLine("signed-pdf-non-pfx requires <input.pdf> <certificate.pem> <private_key.pem>");
                Environment.Exit(1);
                return;
            }

            var apiKey = Environment.GetEnvironmentVariable("PDFREST_API_KEY");
            if (string.IsNullOrWhiteSpace(apiKey)) { Console.Error.WriteLine("Missing required environment variable: PDFREST_API_KEY"); Environment.Exit(1); return; }
            var baseUrl = Environment.GetEnvironmentVariable("PDFREST_URL") ?? "https://api.pdfrest.com";
            var files = args.Take(3).ToArray();
            foreach (var f in files) { if (!File.Exists(f)) { Console.Error.WriteLine($"File not found: {f}"); Environment.Exit(1); return; } }

            using var httpClient = new HttpClient { BaseAddress = new Uri(baseUrl) };
            var ids = new string[files.Length];
            for (int i = 0; i < files.Length; i++)
            {
                var uploadRequest = new HttpRequestMessage(HttpMethod.Post, "upload");
                uploadRequest.Headers.TryAddWithoutValidation("Api-Key", apiKey);
                uploadRequest.Headers.Accept.Add(new("application/json"));
                var bytes = File.ReadAllBytes(files[i]);
                var content = new ByteArrayContent(bytes);
                content.Headers.TryAddWithoutValidation("Content-Type", "application/octet-stream");
                content.Headers.TryAddWithoutValidation("Content-Filename", Path.GetFileName(files[i]));
                uploadRequest.Content = content;
                var uploadResponse = await httpClient.SendAsync(uploadRequest);
                var uploadResult = await uploadResponse.Content.ReadAsStringAsync();
                JObject uploadJson = JObject.Parse(uploadResult);
                ids[i] = uploadJson["files"][0]["id"].ToString();
            }

            var signedPdfRequest = new HttpRequestMessage(HttpMethod.Post, "signed-pdf");
            signedPdfRequest.Headers.TryAddWithoutValidation("Api-Key", apiKey);
            signedPdfRequest.Headers.Accept.Add(new("application/json"));
            signedPdfRequest.Headers.TryAddWithoutValidation("Content-Type", "application/json");

            var signatureConfiguration = new JObject
            {
                ["type"] = "new",
                ["name"] = "esignature",
                ["location"] = new JObject
                {
                    ["bottom_left"] = new JObject { ["x"] = "0", ["y"] = "0" },
                    ["top_right"] = new JObject { ["x"] = "216", ["y"] = "72" },
                    ["page"] = "1"
                },
                ["display"] = new JObject { ["include_datetime"] = "true" }
            };

            JObject parameterJson = new JObject
            {
                ["id"] = ids[0],
                ["certificate_id"] = ids[1],
                ["private_key_id"] = ids[2],
                ["signature_configuration"] = signatureConfiguration.ToString(Formatting.None),
            };
            signedPdfRequest.Content = new StringContent(parameterJson.ToString(), Encoding.UTF8, "application/json");
            var signedPdfResponse = await httpClient.SendAsync(signedPdfRequest);
            var signedPdfResult = await signedPdfResponse.Content.ReadAsStringAsync();
            Console.WriteLine("Processing response received.");
            Console.WriteLine(signedPdfResult);
        }
    }
}

File: Python/Endpoint Examples/JSON Payload/signed-pdf-non-pfx.py
import requests
import json

def upload_file(file_path, file_name):
    with open(file_path, 'rb') as f:
        upload_data = f.read()
        print("Uploading file...")
        upload_response = requests.post(url='https://api.pdfrest.com/upload',
                            data=upload_data,
                            headers={'Content-Type': 'application/octet-stream', 'Content-Filename': file_name, "API-Key": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"})
        print("Upload response status code: " + str(upload_response.status_code))

        if upload_response.ok:
            upload_response_json = upload_response.json()
            print(json.dumps(upload_response_json, indent = 2))

            return upload_response_json['files'][0]['id']
        else:
            print(upload_response.text)
            return ""

input_pdf = '/path/to/input.pdf'
input_id = upload_file(input_pdf, 'input.pdf')

certificate_path = '/path/to/certificate.pem'
certificate_id = upload_file(certificate_path, 'certificate.pem')

private_key_path = '/path/to/private_key.pem'
private_key_id = upload_file(private_key_path, 'private_key.pem')

signature_config = {
    "type": "new",
    "name": "esignature",
    "location": {
        "bottom_left": { "x": "0", "y": "0" },
        "top_right": { "x": "216", "y": "72" },
        "page": 1
    },
    "display": {
        "include_datetime": "true"
    }
}
boxes_data = { "id" : input_id, "certificate_id": certificate_id, "private_key_id": private_key_id, "signature_configuration": json.dumps(signature_config) }

print(json.dumps(boxes_data, indent = 2))


print("Processing file...")
boxes_response = requests.post(url='https://api.pdfrest.com/signed-pdf',
                    data=json.dumps(boxes_data),
                    headers={'Content-Type': 'application/json', "API-Key": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"})



print("Processing response status code: " + str(boxes_response.status_code))
if boxes_response.ok:
    boxes_response_json = boxes_response.json()
    print(json.dumps(boxes_response_json, indent = 2))

else:
    print(boxes_response.text)
